### PR TITLE
Remove batching functionality from cell and pixel mask generation

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Addresses part 2 of #718.

**How did you implement your changes**

Change the iteration in `generate_and_save_{pixel/cell}_cluster_masks` to a per-FOV basis, which requires deprecating the `batch_size` parameter.

`generate_{pixel/cell}_cluster_mask` will also support just a single `fov` string identifying which FOV to process (as opposed to a batch of FOVs).

Because `save_fov_images` and `label_cells_by_cluster` are used elsewhere in the repo, it's best to continue requiring a `fovs` list for these, and instead pass in a list with a single FOV from `generate_{pixel/cell}_cluster_mask`.

### CODE REVIEW

**Comment by ngreenwald** [comment]
Which other functions use save_fov_images and label_cells_by_cluster? Are we going to end up removing the batching functionality from them in the future? Or is it really worth keeping them as is?

**Comment by alex-l-kong** [comment]
@ngreenwald `label_cells_by_cluster` is called in both the post-clustering notebook (via call to `create_mantis_project`) and the neighborhood analysis notebook, both of which pass in a list of FOVs. `save_fov_images` seems just limited to `create_mantis_project`, which also passes in a list of FOVs.

I thought about looping over each FOV in the `create_mantis_project` function, but this doesn't resolve the issue in the neighborhood analysis notebook. I have a few ideas about how to condense this to allow `save_fov_images` and `create_mantis_project` to work on just single FOVs, however it requires some refactoring to be done in the neighborhood analysis notebook itself. Not sure if this is within the scope of this PR, we can address that in a separate PR if need be.

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Okay agreed, that should be a separate PR. As you find stuff like that, go ahead and add it to the overall design doc so we can keep track of what else needs to be changed.

File: ark/utils/data_utils.py [hunk 1]
```diff
@@ -17,11 +17,11 @@
 
 
 def save_fov_images(fovs, data_dir, img_xr, sub_dir=None, name_suffix=''):
-    """Given an xarray of images per fov, saves each image separately
+    """Given an xarray of a fov image, saves it
```
[the remark this answers is not in the document]
Comment (alex-l-kong) [reply ark/utils/data_utils.py:20]: Good catch, seems like leftover from my attempt to make it a per-FOV iteration.

**Review by ngreenwald — APPROVED** [review]

**Pull request merged**